crates/datasources/src/bson/builder.rs

```diff
@@ -51,6 +53,17 @@ impl RecordStructBuilder {
         })
     }
 
+    pub fn len(&self) -> usize {
+        match self.builders.first() {
+            Some(elem) => elem.len(),
+            None => 0,
+        }
+    }
```

Comment:
Semantically, it's difficult for me to understand "what" is being measured here. By looking at the name `RecordStructBuilder`, I would have assumed number of records, but that's not the case here.

Reply:
Oh, I misread this then.